File: schema/src/main/java/dev/sayaya/handbook/usecase/LayoutRepository.java
package dev.sayaya.handbook.usecase;

import dev.sayaya.handbook.domain.Position;
import dev.sayaya.rx.Observable;
import jsinterop.base.JsPropertyMap;

/**
 * 타입 레이아웃 저장소 인터페이스.
 */
public interface LayoutRepository {
    Observable<JsPropertyMap<Position>> load(String workspaceId);
    Observable<Void> save(String workspaceId, JsPropertyMap<Position> positions);
}

File: persist-type/src/main/kotlin/dev/sayaya/handbook/usecase/type/LayoutRepository.kt
package dev.sayaya.handbook.usecase.type

import dev.sayaya.handbook.domain.Layout
import reactor.core.publisher.Mono
import java.util.*

interface LayoutRepository {
    fun findById(workspace: UUID, id: UUID): Mono<Layout>
    fun save(layout: Layout): Mono<Layout>
}